### DISCUSSION TITLE
Setting Up Pocketbase with S3 Storage

### DISCUSSION THREAD

**Discussion by tevintchuinkam** [pocketbase/pocketbase#1981] [Q&A] [answered]
I am trying to run Pocketbase inside a Docker Container on Fly.io and use S3 storage provided my Amazon to store the data. 

 I have existing data (pb_data) from local development that I would like to continue using. I have uploaded this data into the S3 Bucket like so : 
![image](https://user-images.githubusercontent.com/67971452/222973701-10e49739-de13-4fbe-9e10-b63f9f0445e3.png)

And successfully connected the pocketbase instance running in the remote Docker container (I am getting a  " S3 connected successfully" message). 
But no data appears in the dashboard. I guess it is still using the auto-generated database because I am doing something wrong with the configuration.

My Dockerfile looks like this: 

`FROM ubuntu:latest

COPY api/cmd/pocketbase api/cmd/pocketbase

WORKDIR /

RUN apt-get update \
    && apt-get install -y libcap2-bin \
    && apt-get install -y ca-certificates \
    && update-ca-certificates \
    && rm -rf /var/lib/apt/lists/*

RUN setcap \
    'cap_net_bind_service=+ep' \ 
    /api/cmd/familytale

EXPOSE 8080

CMD ["/api/cmd/pocketbase", "serve", "--http=0.0.0.0:8080"]
` 

Thanks a lot for any help in advance!

**Comment by ganigeorgiev** [comment, marked as answer]
The S3 storage is only for the uploaded record files.

The database (`pb_data/data.db`) and the local migrations (`pb_migrations`) must be on the disk.

**Reply by ganigeorgiev** [reply]
You can mount a persistent storage on fly.io using [Volumes](https://fly.io/docs/reference/volumes).

For an example, you can check https://github.com/pocketbase/pocketbase/discussions/537

**Reply by tevintchuinkam** [reply]
Makes sense, thank you very much!
